rfc/rfc-80/rfc-80.md

```diff
@@ -0,0 +1,161 @@
+## Background
+Currently, Hudi organizes data according to fileGroup granularity. The fileGroup is further divided into column clusters to introduce the columnFamily concept.  
+The organizational form of Hudi files is divided according to the following rules:  
+The data in the partition is divided into buckets according to hash; the files in each bucket are divided according to columnFamily; multiple colFamily files in the bucket form a completed fileGroup; when there is only one columnFamily, it degenerates into the native Hudi bucket table.
+
+![table](table.png)
+
+After splitting the fileGroup by columnFamily, the naming rules for base files and log files change. We add the cfName suffix to all file names to facilitate Hudi itself to distinguish column families. The addition of this suffix is compatible with Hudi's original naming method and has no conflict.
+
+![filenames](filenames.png)
+
+## Implementation
+Describe the new thing you want to do in appropriate detail, how it fits into the project architecture. 
+Provide a detailed description of how you intend to implement this feature.This may be fairly extensive and have large subsections of its own. 
+Or it may be a few sentences. Use judgement based on the scope of the change.
+
+### Constraints and Restrictions
+1. The overall design relies on the lock-free concurrent writing feature of Hudi 1.0.  
+2. Lower version Hudi cannot read and write column family tables.  
+3. Only MOR bucketed tables support setting column families.  
+4. Column families do not support repartitioning and renaming.  
+5. Schema evolution does not take effect on the current column family table.  
+6. Like native bucket tables, clustering operations are not supported.
+
+### Model change
+After the column family is introduced, the storage structure of the entire Hudi bucket table changes:
+
+![bucket](bucket.png)
+
+The bucket is divided into multiple columnFamilies by column cluster. When columnFamily is 1, it will automatically degenerate into the native bucket table.
+
+![file-group](file-group.png)
+
+### Specifying column families when creating a table
+In the table creation statement, column family division is specified in the options/tblproperties attribute;
+Column family attributes are specified in key-value mode:  
+* Key is the column family name. Format: hoodie.colFamily. Column family name    naming rules specified.  
+* Value is the specific content of the column family: it consists of all the columns included in the column family plus the precombine field. Format: " col1,col2...colN; precombineCol", the column family list and the preCombine field are separated by ";"; in the column family list the columns are split by ",".  
+
+Constraints: The column family list must contain the primary key, and columns contained in different column families cannot overlap except for the primary key. The preCombie field does not need to be specified. If not specified, the primary key will be taken by default.
```

Comment:
Is there any SQL synctax we can reference in industry? Like the cockroach db: https://www.cockroachlabs.com/docs/stable/column-families#:~:text=A%20column%20family%20is%20a,%2C%20UPDATE%20%2C%20and%20DELETE%20operations.

Reply:
For the first version of this feature I would prefer to use existing abilities as much as possible. I think, if we can setup all we need (column families configuration) using only tblproperties and options - it's better to use them and avoid making any additions/changes (such as new keywords like "family") to Spark/Flink DDL.   
But later, when the feature is done, tested and performed well, if it will be in demand - of cource we could make it more userfriendly by adding approprate SQL syntax.

Reply:
agree with keeping things simple for now, esp given Flink SQL is hard to evolve.